### PR TITLE
Feat: Expenses view 

### PR DESCRIPTION
**Pull request by giovannifavorin**
### Overview
Made Category flow

### Key Changes

- **View**: Made Views for adding, Editing, deleting and seeing data 
- **Model**: Connected with Model, saving data in SwiftData

### Screenshots

Captura de Tela 2024-11-07 às 17 20 32
Captura de Tela 2024-11-07 às 17 20 42
Captura de Tela 2024-11-07 às 17 20 51

### CODE REVIEW

File: MoneyManagement App/Model/ExpenseCategory.swift [hunk 1]
```diff
@@ -7,13 +7,11 @@ final class ExpenseCategory {
     #Unique<ExpenseCategory>([\.id], [])
     
     var id: UUID
-    var idUser: UUID
```
Comment (RafaelOsiro) [inline MoneyManagement App/Model/ExpenseCategory.swift:10]: Please lets follow the Database modeling. If you have some doubts about the modeling we could huddle the dev team to update the database. But do NOT remove the modeling before we have this discussion with the team.

File: MoneyManagement App/View/Views/PaymentMethod/DetailPaymentMethod/Component/EmojiPicker.swift [hunk 2]
```diff
@@ -0,0 +1,60 @@
+//
```
Comment (RafaelOsiro) [inline MoneyManagement App/View/Views/PaymentMethod/DetailPaymentMethod/Component/EmojiPicker.swift:1]: Please remove the comments (lines 1 to 6)

File: MoneyManagement App/View/Views/PaymentMethod/DetailPaymentMethod/DetailPaymentMethod+View.swift [hunk 3]
```diff
@@ -1,27 +1,14 @@
+//
```
Comment (RafaelOsiro) [inline MoneyManagement App/View/Views/PaymentMethod/DetailPaymentMethod/DetailPaymentMethod+View.swift:1]: Please remove the comments (lines 1 to 6)

File: MoneyManagement App/View/Views/PaymentMethod/DetailPaymentMethod/DetailPaymentMethod+ViewModel.swift [hunk 4]
```diff
@@ -0,0 +1,37 @@
+//
```
Comment (RafaelOsiro) [inline MoneyManagement App/View/Views/PaymentMethod/DetailPaymentMethod/DetailPaymentMethod+ViewModel.swift:1]: Please remove the comments (lines 1 to 6)

File: MoneyManagement App/View/Views/PaymentMethod/EditPaymentMethod/EditPaymentMethod+View.swift [hunk 5]
```diff
@@ -1,43 +1,13 @@
+//
```
Comment (RafaelOsiro) [inline MoneyManagement App/View/Views/PaymentMethod/EditPaymentMethod/EditPaymentMethod+View.swift:1]: Please remove the comments (lines 1 to 6)

File: MoneyManagement App/View/Views/PaymentMethod/EditPaymentMethod/EditPaymentMethod+ViewModel.swift [hunk 6]
```diff
@@ -0,0 +1,30 @@
+//
```
Comment (RafaelOsiro) [inline MoneyManagement App/View/Views/PaymentMethod/EditPaymentMethod/EditPaymentMethod+ViewModel.swift:1]: Please remove the comments (lines 1 to 6)

File: MoneyManagement App/View/Views/PaymentMethod/PaymentMethod/PaymentMethod+ViewModel.swift [hunk 7]
```diff
@@ -0,0 +1,15 @@
+//
```
Comment (RafaelOsiro) [inline MoneyManagement App/View/Views/PaymentMethod/PaymentMethod/PaymentMethod+ViewModel.swift:1]: Please remove the comments (lines 1 to 6)

File: MoneyManagement App/View/Views/TransactionCategories/TransactionAlertsTest.swift [hunk 8]
```diff
@@ -0,0 +1,63 @@
+//
```
Comment (RafaelOsiro) [inline MoneyManagement App/View/Views/TransactionCategories/TransactionAlertsTest.swift:1]: Please remove the comments (lines 1 to 6)

**Review by RafaelOsiro — CHANGES_REQUESTED** [review]
Your branch is not updated with develop and have conflicts.

File: MoneyManagement App/Model/ExpenseCategory.swift (hunk 1 above)
Comment (giovannifavorin) [reply MoneyManagement App/Model/ExpenseCategory.swift:10]: Can we schedule a moment to discuss on the necessity of this var? why is it necessary to connect those two data?

**Review by RafaelOsiro — CHANGES_REQUESTED** [review]
I found several bugs that are not following the prototype or the business rules. You might not be aware, but most of the prototype is interactive (this has been the case since I created it), and you could have seen more details by running the Figma preview.
Additionally, the section for adding a financial institution contains a lot of business logic that you could have reviewed as well.

## Bug 1
In the simulator video, when opening a payment method, a sheet (modal) is displayed showing the selected item's details in a read-only state. However, when clicking on "Edit," the data becomes editable. If no changes are made and the user decides to click the back button, the sheet should return to the Read-Only State as shown in the screenshot below. However, clicking the button currently dismisses the sheet. Fix this behavior so that, when editing, if the user makes no changes and clicks back, the sheet DOES NOT DISMISS and instead reverts to its original state.

https://github.com/user-attachments/assets/d671c4f1-2e85-4537-b9c8-c38e3c267501

image

## Bug 2
The hitbox of the cell on the payment methods screen is too small.
This can be observed when I add a red border around the cell's outline.
Only the red-bordered area is selectable, while the rest of the cell is not clickable.

image

## Bug 3
3. The alert message for removing payment methods should be consistent. In **Image 1**, the alert shows both a title and a description. In **Image 2**, it only shows the title. **Image 3** from the prototype indicates that all alerts should be the same, using only the title without a description (both for deletion from the list and deletion when viewing the item).
Simulator Screenshot - iPhone 16 Pro Max - 2024-11-20 at 02 00 02
Simulator Screenshot - iPhone 16 Pro Max - 2024-11-20 at 02 00 07
image

## Bug 4
The slider button in the Earnings list is incorrect. It should only display the trash icon and not the "Delete" label.
Simulator Screenshot - iPhone 16 Pro Max - 2024-11-20 at 02 08 21
image

## Bug 5
The alert to confirm the deletion of an earnings item is missing.

https://github.com/user-attachments/assets/1220aec5-32bc-4168-b175-160e7eecf3a2

image

## Bug 6
When trying to create a new earnings category, if the user does nothing and decides to cancel the addition, the alert should not be displayed.
This alert should only appear if the user is in the middle of editing and clicks the Cancel button.

https://github.com/user-attachments/assets/fde74a67-2488-4f25-8f13-5410880971af

## Bug 7 + Feature Request
The screen's appearance does not match the Figma prototype.
In the Figma screen, there are no rounded borders, and the form items span the entire width of the screen. Additionally, the text field is misaligned with the item above it.
The "Delete Earnings Category" button is not separated; it is grouped with all the other items on the screen.

This is not a bug, but the title can be updated to "Editar Categoria de Ganho". If the user turns off the `Ganho` toggle, the title should change to "Editar Categoria de Gasto".

Simulator Screenshot - iPhone 16 Pro Max - 2024-11-20 at 02 16 36
image

## Bug 8
When clicking the delete button for an item after opening the sheet, the confirmation alert for deletion does not appear.

https://github.com/user-attachments/assets/f3371398-12c1-430e-be32-1f17d9a67e85

image

## Bug 9

1. The user can edit the name and toggle the switch, which should not be allowed in read-only mode.
2. The text and icon should have specific colors in read-only mode (the text opacity differs when adding or editing an earning).
3. The sheet title is incorrect.

https://github.com/user-attachments/assets/fe4cada9-857b-4687-b8e4-0b9d989bb525

image

## Bug 10
When adding an Expense Category, the `Ganho` toggle should be disabled by default when the modal opens.
Similarly, for an Earnings Category, the `Ganho` toggle should be enabled by default when the modal opens.

https://github.com/user-attachments/assets/a9b51e8a-c166-4a7b-b78d-eea276d84d39

image

## Bug 11

1. On both the Earnings and Expense screens, the title size differs from the Figma prototype.
2. On the Expense sheet, the text is incorrect—it is in plural form when it should be singular.

Simulator Screenshot - iPhone 16 Pro Max - 2024-11-20 at 02 40 44

image

image

image

## Bug 12
On the Expenses screen, the same bugs occurring on the Earnings screen are present. Here's the list of issues that need to be fixed for the Expenses section:
Bug 4: The slider button should only display the trash icon and not the "Delete" label.
Bug 5: The confirmation alert for item deletion is missing when clicking the delete button.
Bug 6: In read-only mode, the user can edit fields and toggle switches, which should not be allowed. The text and icon should have appropriate colors for read-only mode.
Bug 7 + Feature Request: The screen title should match the Figma prototype's font size and content. Additionally, if the toggle Earnings is turned off, the title should dynamically update to "Edit Expense Category."
Bug 8: The toggle Earnings should be disabled by default when adding an Expense Category.
Bug 9: The sheet title is in plural form, which is incorrect; it should be singular.

**Pull request closed without merging**